lib/publiccloud/instance.pm

```diff
@@ -66,9 +68,14 @@ sub run_ssh_command {
 
     delete($args{cmd});
     delete($args{no_quote});
+    delete($args{validate});
     delete($args{ssh_opts});
     delete($args{username});
-    return script_output($ssh_cmd, %args);
+    if ($validate == 'no') {
```

Comment:
can you elaborate what you trying to achieve here ?

Reply:
I mean if you don't care about what function returning just don't read it what benefit of explicitly cancel return ?

Reply:
The problem of [`script_output`](http://open.qa/api/testapi/#_script_output) is, that it always waits for the command to exit, but [`script_run()`](http://open.qa/api/testapi/#_script_run) can just execute it and continue, see:
```
If $timeout is greater than 0, wait for that length of time for execution to complete *(otherwise, returns undef immediately)*.
```

Reply:
ah I actually missed that you switching between script_run and script_output , sorry . Make sense